hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java

```diff
@@ -0,0 +1,102 @@
+import static com.hedera.node.app.spi.workflows.HandleException.validateTrue;
+
+import com.hedera.hapi.node.token.CryptoCreateTransactionBody;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.spi.validation.AttributeValidator;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.config.data.CryptoCreateWithAliasConfig;
+import com.hedera.node.config.data.EntitiesConfig;
+import com.hedera.node.config.data.LedgerConfig;
+import com.hedera.node.config.data.TokensConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Provides validation for token fields like token type,  token supply type, token symbol etc.,.
+ * It is used in pureChecks for token creation.
+ */
+@Singleton
+public class CryptoCreateValidator {
+    private static final int EVM_ADDRESS_SIZE = 20;
+
+    @Inject
+    public CryptoCreateValidator() { // Exists for injection
+    }
+
+    /**
+     * Validates Key Alias and EVM Address combinations.
+     *
+     * @param op    the crypto create transaction body
+     * @param attributeValidator   AttributeValidator
+     * @param config    CryptoCreateWithAliasConfig
+     * @param readableAccountStore    ReadableAccountStore
+     * @throws HandleException if the inputs are not invalid
+     */
+    public void validateKeyAliasAndEvmAddressCombinations(
+            @NonNull final CryptoCreateTransactionBody op,
+            @NonNull final AttributeValidator attributeValidator,
+            @NonNull final CryptoCreateWithAliasConfig config,
+            @NonNull final ReadableAccountStore readableAccountStore) {
+        if (op.hasKey() && op.alias().equals(Bytes.EMPTY)) {
+            attributeValidator.validateKey(op.keyOrThrow());
+        } else if (!op.hasKey() && !op.alias().equals(Bytes.EMPTY)) {
+            final var responseCode = config.enabled() ? INVALID_ALIAS_KEY : NOT_SUPPORTED;
+            throw new HandleException(responseCode);
+        } else if (op.hasKey() && !op.alias().equals(Bytes.EMPTY)) {
+            validateKeyAndAliasProvidedCase(op, attributeValidator, config, readableAccountStore);
+        } else {
+            // This is the case when no key and no alias are provided
+            throw new HandleException(KEY_REQUIRED);
+        }
+    }
+
+    private void validateKeyAndAliasProvidedCase(
+            @NonNull final CryptoCreateTransactionBody op,
+            @NonNull final AttributeValidator attributeValidator,
+            @NonNull final CryptoCreateWithAliasConfig config,
+            @NonNull final ReadableAccountStore readableAccountStore) {
+        validateTrue(config.enabled(), NOT_SUPPORTED);
+        attributeValidator.validateKey(op.keyOrThrow());
+        validateTrue(op.alias().toByteArray().length == EVM_ADDRESS_SIZE, INVALID_ALIAS_KEY);
```

Comment:
`isMirror` check seems to be missing here

Reply:
Added. Thanks.